src/main/scala/org/broadinstitute/dsde/firecloud/webservice/NihApiService.scala

```diff
@@ -39,38 +35,7 @@ trait NihApiService extends Directives with RequestBuilding with StandardUserInf
         path("callback") {
           post {
             entity(as[JWTWrapper]) { jwtWrapper =>
-
```

Comment:
I moved all of the JWT decoding into NihService. I preferred having it all contained in the service and it was easier to work with all the Futures in once place

Reply:
Agreed, thanks for moving this!